### PR TITLE
updated train-test splitter to sklearn 0.18

### LINKED ISSUE #71: Update ShuffleLabelsOut to use sklearn.model_selection
Currently, [medleydb.utils.artist_conditional_split](https://github.com/marl/medleydb/blob/master/medleydb/utils.py#L145) relies on the [ShuffleLabelsOut](https://github.com/marl/medleydb/blob/master/medleydb/utils.py#L200) class, which is using the soon to be deprecated sklearn.cross_validation. It needs to be updated to use sklearn.model_selection but things have changed enough that a simple switch breaks the code. @bmcfee I took a stab at this myself but don't understand what you did in the original version well enough to troubleshoot why my changes weren't working. Can you give it a go?

### PR DESCRIPTION
**Pull request by bmcfee**
This should resolve #71 by using sklearn 0.18's built-in splitter.

I also strengthened the unit tests a bit, and added a version requirement to setup.py.

### CODE REVIEW

**Comment by bmcfee** [comment]
Note: coverage % decreased here because I deleted code that was previously covered.

File: setup.py [hunk 1]
```diff
@@ -54,7 +54,8 @@
             'six',
             'librosa',
             'pydrive',
-            'scipy'
+            'scipy',
+            'scikit-learn >= 0.18'
```
Comment (rabitt) [inline setup.py]: whoops! didn't realize I was missing the scikit-learn dependency all together :/

File: tests/test_utils.py [hunk 2]
```diff
@@ -175,3 +184,13 @@ def test_artist_index(self):
             sorted(splits[0]['train'] + splits[0]['test'])
         )
 
+        # Same tests as above
+        self.assertEqual(set(splits[0]['train']) & set(splits[0]['test']),
+                         set())
+        self.assertNotEqual(len(splits[0]['train']), 0)
+        self.assertNotEqual(len(splits[0]['test']), 0)
+
+        # Also make sure that artists are partitioned properly
```
Comment (rabitt) [inline tests/test_utils.py]: Thanks for adding these 👍

**Comment by rabitt** [comment]
thanks @bmcfee
